prober/icmp.go

```diff
@@ -136,3 +132,96 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 		}
 	}
 }
+
+func getICMPSocket(
```

Comment:
I don't think you're buying much by breaking this out, and it also makes it more difficult to review what has changed in the code